services/src/main/java/org/keycloak/services/DefaultKeycloakSessionFactory.java

```diff
@@ -59,6 +59,8 @@
     // TODO: Likely should be changed to int and use Time.currentTime() to be compatible with all our "time" reps
     protected long serverStartupTimestamp;
 
+    private long clientStorageProviderTimeout;
```

Comment:
We should add a javadoc for the timeout. What it does and in what units it's expected. Not sure where tha javadoc should be, whether in `DefaultKeycloakSessionFactory` or `DefaultKeycloakSession` or `ClientStorageManager`. Maybe  everywhere? :)

Reply:
I believe `DefaultKeycloakSessionFactory` is the right place for the time being, since the configuration option should only be parsed once rather than with every `DefaultKeycloakSession` instantiation.

Sadly, `ClientStorageManager` is not instantiated as other SPIs so lacks ability to be configured in the init() method, and we have to simulate it from here. In the future, once we refactor the `ClientStorageManager` to be more SPI-like, it will be moved accordingly.

Reply:
done